### PR TITLE
Adjust event descriptions for collections.

### LINKED ISSUE #2303: Add additional descriptions to collections history table for some events and update one description
Some user actions are not generating descriptions in the right-most column of the collections history table on the view page. Some of these are specific to when a user saves a draft or deposits, and sometimes it needs to be added for only one of these (because it works fine now for the other button). I would like to have the following added in:

- when contact is removed and item is saved as a draft: "contact email modified"
- when related link/link text are removed and collection is saved as draft OR deposited: "related links modified"
- when release setting is changed: "release settings modified"
- when download (visibility) setting is changed: "download setting modified"
- when DOI assignment setting is changed: DOI setting modified"
- when license setting is changed: "license setting modified"
- when either or both of the boxes for notifications to managers or collection participants are checked or unchecked: "notification settings mnodified"
- when review workflow is turned on or off: "review workflow setting modified". 

Note that currently if there are reviewers and the review workflow is turned off, the description says "Removed reviewers: [SUNet ID]". In this case, this text should remain and the above "review workflow setting modified" text should appear as well.

Also, when a related link is added, for some reason the descriptive text says both "related links modified" and "contact email modified". The comment about the contact email should be removed.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #2303

## Why was this change made? 🤔
Better event descriptions.

## How was this change tested? 🤨

⚡ ⚠ If this change involves consuming from or writing to another service (or shared file system), ***run [integration test create_object_h2_spec.rb](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test manually in [stage|qa] environment, in addition to specs. ⚡

Unit

### CODE REVIEW

File: spec/services/collection_change_set_spec.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+require 'rails_helper'
+
+RSpec.describe CollectionChangeSet do
+  let(:collection1) { create(:collection) }
+  let(:collection2) do
+    create(:collection, email_when_participants_changed: true, email_depositors_status_changed: true,
+                        review_enabled: true)
+  end
+
+  describe '#email_depositors_status_changed_changed?' do
```
Comment (ndushay) [inline spec/services/collection_change_set_spec.rb:12]: changed_changed?

File: app/models/collection_change_set.rb [hunk 2]
```diff
@@ -44,6 +44,22 @@ def changed_participants
       removed_reviewers).uniq
   end
 
+  def email_depositors_status_changed_changed?
```
Comment (ndushay) [inline app/models/collection_change_set.rb:47]: changed_changed?

**Review by ndushay — COMMENTED** [review]
I would think Amy wants to review this, if she hasn't already.   A couple of concerns for your consideration.

File: spec/services/collection_change_set_spec.rb (hunk 1 above)
Comment (justinlittman) [reply spec/services/collection_change_set_spec.rb:12]: Unfortunately. The field is called `email_depositors_status_changed` so checking whether this field has changed appends `_changed?`.

File: app/models/collection_change_set.rb (hunk 2 above)
Comment (justinlittman) [reply app/models/collection_change_set.rb:47]: Unfortunately. The field is called `email_depositors_status_changed` so checking whether this field has changed appends `_changed?`.

**Comment by justinlittman** [comment]
@amyehodge did not request to review this.

**Review by jcoyne — APPROVED** [review]

**Pull request merged**